### PR TITLE
Send redirectOnError param to getPublishedContent when token is pulled from cookie

### PR DESCRIPTION
**Pull request by spastorelli**
This PR aims to update GBO after changes made in https://github.com/GitbookIO/gitbook-x/pull/16138

### CODE REVIEW

File: packages/gitbook/src/middleware.ts [hunk 1]
```diff
@@ -685,14 +686,23 @@ async function lookupSiteByAPI(
         `lookup content for url "${url.toString()}", with ${lookup.urls.length} alternatives`,
     );
 
+    // When the cookie is pulled from the cookie, set redirectOnError when calling getPublishedContentByUrl to allow
```
Comment (scazan) [inline packages/gitbook/src/middleware.ts:689]: Is this meant to be `When the token is pulled from the cookie,`

Comment (emmerich) [inline packages/gitbook/src/middleware.ts:689]:
```suggestion
    // When the token is pulled from the cookie, set redirectOnError when calling getPublishedContentByUrl to allow
```

File: packages/gitbook/src/middleware.ts [hunk 2]
```diff
@@ -685,14 +686,23 @@ async function lookupSiteByAPI(
...
+    // redirecting when the token is invalid as we could be dealing with stale token stored in the cookie.
+    // For example when the VA backend signature has changed but the token stored in the cookie is not yet expired.
+    const redirectOnError =
+        typeof visitorAuthToken !== 'undefined' && isVisitorAuthTokenFromCookies(visitorAuthToken)
+            ? true
```
Comment (emmerich) [inline packages/gitbook/src/middleware.ts:694]:
```suggestion
```

Not needed, can just be
```typescript
const redirectOnError = typeof visitorAuthToken !== 'undefined' && isVisitorAuthTokenFromCookies(visitorAuthToken)
```

File: packages/gitbook/src/middleware.ts [hunk 3]
```diff
@@ -685,14 +686,23 @@ async function lookupSiteByAPI(
...
+            : false;
+
     const result = await race(lookup.urls, async (alternative, { signal }) => {
         const data = await getPublishedContentByUrl(
             alternative.url,
             typeof visitorAuthToken === 'undefined'
```
Comment (scazan) [inline packages/gitbook/src/middleware.ts:700]: More just a curiosity, but why check typeof instead of `=== undefined` ?

Comment (scazan) [reply packages/gitbook/src/middleware.ts:700]: ah this is from previous code anyhow.

**Review by scazan — APPROVED** [review]

File: packages/gitbook/src/middleware.ts (hunk 1 above)
Comment (spastorelli) [reply packages/gitbook/src/middleware.ts:689]: Yep thanks

File: packages/gitbook/src/middleware.ts [hunk 4]
```diff
@@ -685,14 +686,23 @@ async function lookupSiteByAPI(
         `lookup content for url "${url.toString()}", with ${lookup.urls.length} alternatives`,
     );
 
+    // When the cookie is pulled from the cookie, set redirectOnError when calling getPublishedContentByUrl to allow
+    // redirecting when the token is invalid as we could be dealing with stale token stored in the cookie.
+    // For example when the VA backend signature has changed but the token stored in the cookie is not yet expired.
+    const redirectOnError =
+        typeof visitorAuthToken !== 'undefined' && isVisitorAuthTokenFromCookies(visitorAuthToken)
```
[the remark this answers is not in the document]
Comment (spastorelli) [reply packages/gitbook/src/middleware.ts:693]: Thanks 👍

File: packages/gitbook/src/middleware.ts (hunk 2 above)
Comment (spastorelli) [reply packages/gitbook/src/middleware.ts:694]: Thanks 👍

**Pull request merged**